tests/e2e/e2e_test.go

```diff
@@ -34,8 +34,8 @@ const (
 	// The amount of time to wait after a blockchain tx to make sure that the a block with the
 	// tx was produced and the indexer has indexed the block.
 	// The node seems to generate a block per second.
-	// 2 seconds is experimentally enough, 1 second was flaky.
-	indexerDelay = 2 * time.Second
+	// 6 seconds is experimentally enough.
+	indexerDelay = 6 * time.Second
```

Comment:
Since the indexer now tries to index past the current height (see changes to block.go), the indexer fails often and thus hits the max backoff during the e2e tests while waiting for the node to produce blocks. Changed to 6 seconds here to be in line with the max backoff.

Reply:
It's not totaly clear to me why this happens in case `slowSync=true` - which is currently always the case. I think the backoff-logic is roughly same as it was before #389: https://github.com/oasisprotocol/oasis-indexer/blob/849bf5a8d91406a846de96b41d7c44c934518b5a/analyzer/consensus/consensus.go#L137-L166

In slow-sync mode it will stop processing the batch after the first failure so I would assume there would be a similar number of `backoff.Failure` calls as in the old code. 

I probably missed something.

(it is clear why this would be problematic in fast-sync mode, however that's not used at the moment).

Reply:
> To me, it looks like we're not calling backoff.Success() frequently enough

I think we call it after every successfully processed block, as before?

> In any case, 2 seconds consistently failed during my testing, and I think bumping it up to 6 seconds is the most future-proof value for now since it's in line with the max backoff. (Also since the e2e_test takes 4-5min to run anyway the additional 4 seconds is relatively minor).

Agreed that increasing the test timeout is minor, but this also indicates that the indexer will be ~6 seconds behind when processing at the tip of the chain. Which previously was not the case.

---

Anyway, I found out the reason for the regression. In old code, we waited for "current-timeout" after processing every block. 
With the updated code, we only wait for the timeout period when picking a batch. And because we pick 1000 future blocks we are almost guaranteed to fail mid batch. This, together with failures being more punishable than successes rewarded then leads to this error.

I would suggest doing this:

```golang
		for _, height := range heights {
			// If running in slow-sync, we are likely at the tip of the chain and are picking up
			// blocks that are not yet available. In this case, wait before processing every block,
			// so that the backoff mechanism can tweak the per-block wait time as needed.
			if b.slowSync {
				select {
				case <-time.After(backoff.Timeout()):
					// Process another batch of blocks.
				case <-ctx.Done():
					b.logger.Warn("shutting down block analyzer", "reason", ctx.Err())
					return
				}
			}

			b.logger.Info("processing block", "height", height)
			// Rest of the code...


```

I'm almost certain that with this change you should be able to revert the `indexerDelay` change, but have not explicitly tested it myself.

No hard feelings if you decide to disregard the suggestion, though.

Thanks!

Reply:
Thanks for digging into that! Happy to include your suggestion; will test it and update

Reply:
Nice. Just in case it doesn't work out, an alternative (and probably equivalent) approach is to move `blocksBatchSize` from being a constant to being a property of `blockAnalyzer`, and setting it to 1 in the constructor when `slowSync` is enabled.

Reply:
Looks like @ptrus 's suggestion worked; I'll leave the `blocksBatchSize` change for a potential future PR :)

Reply:
Thanks!